src/instanceMethods/get-template-params.js

```diff
@@ -0,0 +1,109 @@
+import defaultParams from '../utils/params.js'
+import { toArray, capitalizeFirstLetter } from '../utils/utils.js'
+
+export const getTemplateParams = (params) => {
+  const template = typeof params.template === 'string' ? document.querySelector(params.template) : params.template
+  if (!template) {
+    return {}
+  }
+  const templateContent = template.content || template // IE11
+
+  const result = Object.assign(
+    getSwalParams(templateContent),
+    getSwalButtons(templateContent),
+    getSwalImage(templateContent),
+    getSwalIcon(templateContent),
+    getSwalInput(templateContent),
+    getSwalStringParams(templateContent, ['title', 'html', 'footer']),
+  )
+  return result
+}
+
+const getSwalParams = (templateContent) => {
+  const result = {}
+  toArray(templateContent.querySelectorAll('swal-param')).forEach((param) => {
+    const paramName = param.getAttribute('name')
+    let value = param.getAttribute('value')
+    if (typeof defaultParams[paramName] === 'boolean' && value === 'false') {
+      value = false
+    }
+    if (typeof defaultParams[paramName] === 'object') {
+      value = (new Function(`return ${value}`))() // eslint-disable-line no-new-func
```

Comment:
I think there may be CSP (content security policy) issues (similar to #1926) from using the `Function` constructor here.  
I think I know why you opted for this instead of using `JSON.parse`.. `JSON.parse` is too strict, right? (property names must be quoted.. strings and property names must use double quotes, etc)
Not sure what to suggest as a solution here. 
My first idea was to use a package like [json5](https://www.npmjs.com/package/json5), which could be built into sweetalert2 (so still no external dependencies) but that would add a whopping 31kb. https://bundlephobia.com/result?p=json5@2.1.3
I searched for a more lightweight alternative but had no luck.
This might be an issue that doesn't need to be solved. The effect would just be some false-positive security alerts from security analysis tools. There's no real vulnerability unless for some crazy reason the developer is passing user-provided data for these options (I can't imagine why someone would do that).

Reply:
Good catch! In some environments, such as Google Chrome Apps, this will indeed violate CSP.

> I think I know why you opted for this instead of using JSON.parse.. JSON.parse is too strict, right? (property names must be quoted.. strings and property names must use double quotes, etc)

Correct. But this inconvenience seems to be a lesser evil than violating CSP or adding a JSON parser.

Reply:
Replace by `JSON.parse` 👍